dashboard/lib/services/lti.rb

```diff
@@ -192,6 +192,16 @@ def self.sync_section_roster(lti_integration, lti_section, nrps_section)
           nrps_member: nrps_member
         )
         user_was_new = user.new_record?
+        # Update name if different from the NRPS response
+        unless user_was_new
+          nrps_member_message = Policies::Lti.issuer_accepts_resource_link?(issuer) ? nrps_member[:message].first : nrps_member
+          if user.user_type == ::User::TYPE_TEACHER
```

Comment:
as an option, we can use the method `user.teacher?`
https://github.com/code-dot-org/code-dot-org/blob/fe3dc17d082b477d7985b3d331f2fc179b498047/dashboard/app/models/user.rb#L1515-L1521